### PR TITLE
nginx: Added building of nginx-ssl (nginx with SSL support)

### PR DESCRIPTION
**Pull request by Xotic750**
Signed-off-by: Graham Fairweather xotic750@gmail.com

### CODE REVIEW

**Comment by blogic** [comment]
@Xotic750  - the SoB line is bad
should be -> Signed-off-by: Graham Fairweather <xotic750@gmail.com>

**Comment by Xotic750** [comment]
@blogic Hi, I don't understand what is wrong with it, could you be more specific? The source looks like this

`Signed-off-by: Graham Fairweather <xotic750@gmail.com>`

File: net/nginx/Makefile [hunk 1]
```diff
@@ -17,179 +17,202 @@ PKG_MD5SUM:=088292d9caf6059ef328aa7dda332e44
...
 endif
-ifneq ($(CONFIG_NGINX_HTTP_UWSGI),y)
+ifneq ($(CONFIG_NGINX_$(BUILD_VARIANT)_HTTP_UWSGI),y)
   ADDITIONAL_MODULES += --without-http_uwsgi_module
 endif
-ifneq ($(CONFIG_NGINX_HTTP_SCGI),y)
+ifneq ($(CONFIG_NGINX_$(BUILD_VARIANT)_HTTP_SCGI),y)
   ADDITIONAL_MODULES += --without-http_scgi_module
 endif
 ifneq ($(CONFIG_NGINX_HTTP_MEMCACHED),y)
```
Comment (Xotic750) [inline net/nginx/Makefile]: Missing BUILD_VARIANT

File: net/nginx/Makefile [hunk 2]
```diff
@@ -17,179 +17,202 @@ PKG_MD5SUM:=088292d9caf6059ef328aa7dda332e44
...
   ADDITIONAL_MODULES += --without-http_memcached_module
 endif
-ifneq ($(CONFIG_NGINX_HTTP_LIMIT_CONN),y)
+ifneq ($(CONFIG_NGINX_$(BUILD_VARIANT)_HTTP_LIMIT_CONN),y)
   ADDITIONAL_MODULES += --without-http_limit_conn_module
 endif
-ifneq ($(CONFIG_NGINX_HTTP_LIMIT_REQ),y)
+ifneq ($(CONFIG_NGINX_$(BUILD_VARIANT)_$(BUILD_VARIANT)_HTTP_LIMIT_REQ),y)
```
Comment (Xotic750) [inline net/nginx/Makefile]: Has BUILD_VARIANT twice.

**Comment by Xotic750** [comment]
Should there also be a `PKG_CONFIG_DEPENDS` definition for the SSL and non-SSL version to prevent them from both being selected/installed?

**Comment by diizzyy** [comment]
@Xotic750 
Can you please refresh this patch or close it if there's no interest?
Are you still wondering about PKG_CONFIG_DEPENDS usage?

**Comment by Xotic750** [comment]
Maybe at some point in the future but not just now, thanks.

**Pull request closed without merging**